src/components/UsersList/UsersList.jsx

```diff
@@ -0,0 +1,37 @@
+import { List, ListItem, Typography, Box } from '@mui/material';
+import { UsersPosts } from '../UsersPosts/UsersPosts';
+
+export const UsersList = ({ users, posts, comments }) => {
+  const getUserPostsCount = (userId) => {
+    return posts.filter((post) => post.userId === userId).length;
```

Comment:
You're filtering all posts for each user. This has O(n^2) complexity.
If you prepare your data structure beforehand, you won't have to do that.